### PR TITLE
KPI token realm

### PR DESCRIPTION
**Pull request by rcillo**
The realm is needed to distinguish an employee from an application.

https://techjira.zalando.net/browse/ARUHA-1406

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/security/ClientResolver.java [hunk 1]
```diff
@@ -50,15 +49,25 @@ public Client resolveArgument(final MethodParameter parameter,
...
-    private Set<String> getScopes() {
+    public String getRealm() {
+        String realmString = "";
+
         final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-        if (authentication instanceof OAuth2Authentication) {
-            return ((OAuth2Authentication) authentication).getOAuth2Request().getScope();
+        if (OAuth2Authentication.class.equals(authentication.getClass())) {
+            final Object details = ((OAuth2Authentication) authentication).getUserAuthentication().getDetails();
+            if (details instanceof Map) {
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/security/ClientResolver.java]: do you think we could avoid 2 additional inference of type here? it is gonna be done on a lot requests, but does it really necessary?

Comment (rcillo) [reply src/main/java/org/zalando/nakadi/security/ClientResolver.java]: That sounds like a good idea. I'll give it a try.

File: src/main/java/org/zalando/nakadi/security/ClientResolver.java [hunk 2]
```diff
@@ -15,9 +15,8 @@
 import org.zalando.nakadi.util.FeatureToggleService;
 
 import java.security.Principal;
-import java.util.Collections;
+import java.util.Map;
 import java.util.Optional;
-import java.util.Set;
 
 import static org.zalando.nakadi.config.SecuritySettings.AuthMode.OFF;
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/security/ClientResolver.java]: could you please remove the injection of unused FeatureToggleService? thank you

File: src/main/java/org/zalando/nakadi/security/ClientResolver.java (hunk 1 above)
Comment (rcillo) [reply src/main/java/org/zalando/nakadi/security/ClientResolver.java]: What do you think about a5c3ded

Comment (rcillo) [reply src/main/java/org/zalando/nakadi/security/ClientResolver.java]: What do you think about a5c3ded

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
deploy validation please

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
👍

**Comment by rcillo** [comment]
deploy validation please

File: src/main/java/org/zalando/nakadi/security/ClientResolver.java (hunk 2 above)
Comment (rcillo) [reply src/main/java/org/zalando/nakadi/security/ClientResolver.java]: Just for the record, it was removed 👍

**Comment by rcillo** [comment]
👍

**Comment by adyach** [comment]
👍

**Comment by rcillo** [comment]
👍

**Comment by rcillo** [comment]
👍

**Comment by adyach** [comment]
👍

**Pull request merged**